Library/Homebrew/cmd/update-report.rb

```diff
@@ -302,6 +280,83 @@ def link_completions_manpages_and_docs(repository = HOMEBREW_REPOSITORY)
   def migrate_gcc_dependents_if_needed
     # do nothing
   end
+
+  def analytics_message
+    if !Utils::Analytics.messages_displayed? &&
+       !Utils::Analytics.disabled? &&
+       !Utils::Analytics.no_message_output?
+
+      ENV["HOMEBREW_NO_ANALYTICS_THIS_RUN"] = "1"
+      # Use the shell's audible bell.
+      print "\a"
+
+      # Use an extra newline and bold to avoid this being missed.
+      ohai "Homebrew has enabled anonymous aggregate formula and cask analytics."
+      puts <<~EOS
+        #{Tty.bold}Read the analytics documentation (and how to opt-out) here:
+          #{Formatter.url("https://docs.brew.sh/Analytics")}#{Tty.reset}
+        No analytics have been recorded yet (nor will be during this `brew` run).
+
+      EOS
+
+      # Consider the messages possibly missed if not a TTY.
+      Utils::Analytics.messages_displayed! if $stdout.tty?
+    elsif Utils::Analytics.disabled?
+      ohai "Homebrew's analytics are moving to a InfluxDB instance in the EU."
+      puts "Please reconsider disabling just Google Analytics instead of all analytics with:"
+      puts "  brew analytics on"
+      puts "  export HOMEBREW_NO_GOOGLE_ANALYTICS=1\n\n"
+
+      # Consider the message possibly missed if not a TTY.
+      Settings.write "influxanalyticsmessage", true if $stdout.tty?
+    end
+  end
+
+  def donation_message
+    return if Settings.read("donationmessage") == "true"
+
+    ohai "Homebrew is run entirely by unpaid volunteers. Please consider donating:"
+    puts "  #{Formatter.url("https://github.com/Homebrew/brew#donations")}\n\n"
+
+    # Consider the message possibly missed if not a TTY.
+    Settings.write "donationmessage", true if $stdout.tty?
+  end
+
+  def install_from_api_message
+    return if Settings.read("installfromapimessage") == "true"
+
+    auto_update_secs_set = Homebrew::EnvConfig.api_auto_update_secs.to_i.positive?
+    if !Homebrew::EnvConfig.no_auto_update? && !Homebrew::EnvConfig.no_install_from_api? && !auto_update_secs_set
+      return
+    end
+
+    ohai "You have set:"
+    puts "  HOMEBREW_NO_AUTO_UPDATE" if Homebrew::EnvConfig.no_auto_update?
+    puts "  HOMEBREW_API_AUTO_UPDATE_SECS" if auto_update_secs_set
+    if Homebrew::EnvConfig.no_install_from_api? && !automatically_set_no_install_from_api?
+      puts "  HOMEBREW_NO_INSTALL_FROM_API"
+    end
+    puts "but we have dramatically sped up and fixed many bugs in the way we do Homebrew updates since."
+    puts "Please consider unsetting these and tweaking the values based on the new behaviour."
+    puts "\n\n"
+
+    # Consider the message possibly missed if not a TTY.
+    Settings.write "installfromapimessage", true if $stdout.tty?
+  end
+
+  def untap_message
+    return if Homebrew::EnvConfig.no_install_from_api?
+    return if Homebrew::EnvConfig.developer? || ENV["HOMEBREW_DEV_CMD_RUN"]
+
+    core_tap = CoreTap.instance
+    cask_tap = Tap.default_cask_tap
+    return if !core_tap.installed? && !cask_tap.installed?
```

Comment:
Happy with this after Bo's correction.

Mild suggestion:

Maybe we can find a way to limit the number of times this message is displayed in a given time period? I'm thinking of users who generally install from the API, don't run dev commands, but will occasionally build a formula from source because they want to tweak the build.

If it's too complicated then I'm fine with showing it all the time, but it would be nice to not have to.

Reply:
@carlocab How about we leave as-is and I'll make a `critical` fix if people complain? Alternatively, we use `Homebrew::EnvConfig.no_env_hints?` to disable this messaging?

Reply:
The former works for me! Putting it with `HOMEBREW_NO_ENV_HINTS` is a little confusing since these aren't env variables.